Makefile

```diff
@@ -227,6 +227,9 @@ check-yaml:
 check-gosec: gosec
 	$(GOSEC) -quiet -exclude=G101 -fmt json ./...
 
+check-gitlint: gitlint
+	$(GITLINT) -C .gitlint --commits HEAD~10..HEAD lint
```

Comment:
Using `gitlint --commits master..` was my first option, but it turned out that CI checkout action does not have branches, and therefore I used the last 10 commits (which is indeed, arbitrary). I will try to dive deeper into the checkout action code and see what can be done in that respect.

Reply:
If github's being difficult I wonder if something in the https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads visible in their template could work

Totally untested: `make BASE_COMMIT=${ github.events.before }`  or one of the other "commits" related values?

Maybe we should push a temporary step in the action to `git log` so we have a better idea of what trickery github is doing when it checks out a pr for an actions run?

Reply:
@phlogistonjohn I tried all kind of strategies but could not find a proper solution. It appears that the checkout action is doing something unusual. When adding `git rev-list master..` it looks like all commits are listed. See: https://github.com/synarete/samba-operator/actions/runs/4004661629/jobs/6874068895

Reply:
OK, that's a big disappointment (in github not your effort). My concrete suggestion is to increase the count to 20 to match with number of commits noted in https://github.com/ceph/go-ceph/blob/master/docs/development.md : 
```
commit sarray of objects Required

An array of commit objects describing the pushed commits. (Pushed commits are all commits that are included in the compare between the before commit and the after commit.) The array includes a maximum of 20 commits. If necessary, you can use the [Commits API](https://docs.github.com/rest/reference/repos#commits) to fetch additional commits. This limit is applied to timeline events only and isn't applied to webhook deliveries.
```

And we can add a short comment saying we chose 20 to match with this number.

In the future I may take a stab at figuring out a more dynamic way of doing it but I won't hold up this PR on that just because github wants to be special.

Reply:
The truth is, that this experience made we wonder if `gitlint` is the right tool for SINK, at least with respect to CI workflow. For example, running `gitlint -C .gitlint --commits HEAD~100..HEAD lint` defines the following commits as non-valid:

https://github.com/samba-in-kubernetes/samba-operator/commit/69e26ff28edb88eac66e3723f09940503e29c097

https://github.com/samba-in-kubernetes/samba-operator/commit/fd5ef5013babb695843d85cf45d6b87d5af4a9e6

Maybe just have a `make` rule?